x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -4,6 +4,14 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import {
+  BasePath,
+  SavedObjectsClient,
+  KibanaRequest,
+  SavedObjectsClientContract,
+  // @TODO: Move server deps to /server
```

Comment:
Just a note about this: I think I'd like to have all of these definitions moved to `server/types.ts` in the new platform. Some of our type definition files are named `types.d.ts`. It doesn't break anything, but if somehow one of these types is exported from a public JS file, it will break compilation of the bundles at build time in CI. 

Just a bit of what I've been working on along the side of this PR :)